### PR TITLE
Adjust DynamicDictionaryValue TryParse method to allow more convertions

### PR DESCRIPTION
**Pull request by khalidsalomao**
Adjust DynamicDictionaryValue TryParse method to allow more convertions.
. Convertions like Int64 to Int32 were ignored in the previous version,
since there was no ending else statement.
Now the last rule is to try a type convertion.
. Convertions to a nullable type were also ignored (like Int64 to
Int64?), now there is a test to check if a nullable type convertion is
possible.

Note that if the convertion fails any exception will be treated and the
default value returned.

Some test cases were created for these situations.

### CODE REVIEW

File: src/Nancy.Tests/Unit/DynamicDictionaryValueFixture.cs [hunk 1]
```diff
@@ -898,6 +898,66 @@ public void Should_implicitly_convert_from_string_based_on_given_generic_type_of
         }
 
         [Fact]
+        public void Should_implicitly_convert_from_int_based_on_given_parameter_type_of_short ()
```
Comment (horsdal) [inline src/Nancy.Tests/Unit/DynamicDictionaryValueFixture.cs]: Nitpickng: There is usually no spaces before the parenthesis in Nancy code.

**Comment by horsdal** [comment]
:+1:

**Comment by khalidsalomao** [comment]
Hi Horsdal, 

One more thing about DynamicDictionaryValue: the methods "TryParse" and "TryConvert" share some similarities.... Should I try to merge the common funcionalities of both methods? Or do you guys prefer to leave as it is?

best!

**Comment by horsdal** [comment]
Hmm, not sure they are similar enough for unification.

**Comment by khalidsalomao** [comment]
OK!

**Comment by khellang** [comment]
Love it! :heart:

**Pull request merged**